tensorrt_llm/serve/visual_gen_utils.py

```
@@ -90,21 +90,147 @@ def _merge_extra_params(
+        ref_path = _materialize_reference(
+            payload,
+            modality=modality,
+            ref_id=f"{id}_{modality}_ref_{i}",
+            media_storage_path=media_storage_path,
+        )
+        refs.append(MediaRef(content=ref_path, role=role))
+    return refs
+
+
+def _apply_deprecated_input_reference(
+    input_reference, params, *, id: str, media_storage_path: Optional[str]
```

Ruff (0.16.1)

[error] Function argument `id` is shadowing a Python builtin

(A002)